docs/site/tutorials/todo-list/todo-list-tutorial-repository.md

```diff
@@ -85,6 +85,133 @@ export class TodoListRepository extends DefaultCrudRepository<
 }
 ```
 
+### Inclusion of Related Models
+
+To get the related `Todo` object for each `TodoList`, we have to override the
+`find` and `findById` functions.
+
+First add the following imports:
+
+```ts
+import {Filter, Options} from '@loopback/repository';
+import {TodoListWithRelations} from '../models';
+```
+
+Add the following two functions after the constructor:
+
+{% include code-caption.html content="src/repositories/todo-list.repository.ts" %}
+
+```ts
+async find(
+  filter?: Filter<TodoList>,
+  options?: Options,
+): Promise<TodoListWithRelations[]> {
+  // Prevent juggler for applying "include" filter
+  // Juggler is not aware of LB4 relations
+  const include = filter && filter.include;
+  filter = filter && Object.assign(filter, {include: undefined});
+  const result = await super.find(filter, options);
+
+  // poor-mans inclusion resolver, this should be handled by DefaultCrudRepo
+  // and use `inq` operator to fetch related todos in fewer DB queries
+  if (include && include.length && include[0].relation === 'todos') {
+    await Promise.all(
+      result.map(async r => {
+        r.todos = await this.todos(r.id).find();
+      }),
+    );
+  }
+
+  return result;
+}
+
+async findById(
+  id: typeof TodoList.prototype.id,
+  filter?: Filter<TodoList>,
+  options?: Options,
+): Promise<TodoListWithRelations> {
+  // Prevent juggler for applying "include" filter
+  // Juggler is not aware of LB4 relations
+  const include = filter && filter.include;
+  filter = filter && Object.assign(filter, {include: undefined});
```

Comment:
Ditto, please don't modify the original `filter` object.

Same comment applies to other examples below.